Casks/mini-program-studio.rb

```diff
@@ -1,12 +1,17 @@
 cask "mini-program-studio" do
-  version "1.15.7,b9179b0f-9656-4f2a-a42d-f08e335dd696"
-  sha256 "1a0d1e322a36a83e1d887d569efcdb0a88f0edec786a60301599d9c99e150b8c"
+  version "2.0.6,45f57ec9-2dd1-4150-b107-2003b55b91d6"
+  sha256 "d825eed5500d8127741c4b691826ff06b54e66a6f583de7bbbb5a4ffe5672058"
 
-  url "https://gw.alipayobjects.com/os/bmw-prod/#{version.after_comma}.dmg",
+  url "https://gw.alipayobjects.com/os/volans-demo/#{version.after_comma}/MiniProgramStudio-#{version.before_comma}.dmg",
       verified: "gw.alipayobjects.com/"
   name "Mini Program Studio"
   name "小程序开发者工具"
-  homepage "https://open.alipay.com/channel/miniIndex.htm"
+  desc "DingTalk, produced by Alibaba, is a seamless communication across devices"
+  homepage "https://opendocs.alipay.com/mini/ide"
+
+  livecheck do
+    skip "The content of https://opendocs.alipay.com/mini/ide/download is dynamic"
+  end
```

Comment:
```suggestion
  # The download page (https://opendocs.alipay.com/mini/ide/download) is
  # rendered using JavaScript. The URL below returns JSON data that contains
  # the HTML for the release information, so we can match within that.
  livecheck do
    url "https://opendocs.alipay.com/api/content/006l6m"
    regex(%r{href=.*?/([a-z\d-]+)/MiniProgramStudio[._-]v?(\d+(?:\.\d+)+)\.dmg}i)
    strategy :page_match do |page|
      page.scan(regex).map do |match|
        (match[0] && match[1]) ? "#{match[1]},#{match[0]}" : ""
      end
    end
  end
```

Here we are. The JSON data at the `/api/content/006l6m` endpoint has a `text` field that contains the release information HTML, so we can match within that like we normally would on an HTML page. To format the version information like the cask, I've used a regex with two capture groups and a `strategy` block that assembles the captured text accordingly.

Reply:
Thanks